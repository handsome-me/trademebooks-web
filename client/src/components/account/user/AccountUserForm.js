import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { MDBInput, MDBBtn } from 'mdbreact'
import { toastr } from 'react-redux-toastr'

const AccountUserForm = ({ auth: { user }, updateAuthUser }) => {
  const [formData, setFormData] = useState({
    first_name: user.first_name,
    last_name: user.last_name
  })

  const { first_name, last_name } = formData

  const onChange = (e) => {
    setFormData({ ...formData, [e.target.name]: e.target.value })
  }

  const updateUser = async (e) => {
    e.preventDefault()

    const response = await updateAuthUser({ first_name, last_name })

    if (response) {
      toastr.success(
        'Your name has been changed.',
        'To see your name change be taken into effect, logout then log back in.',
        { timeOut: 0 }
      )
    }
  }

  return (
    <>
      <h3 className="mb-4 font-weight-bold">User Settings</h3>
      <div className="mt-5">
        <h4>Change Name</h4>
        <MDBInput
          label="First Name"
          icon="user"
          group
          type="text"
          name="first_name"
          onChange={onChange}
          required
          value={first_name}
        />
        <MDBInput
          label="Last Name"
          icon="user"
          group
          type="text"
          name="last_name"
          onChange={onChange}
          required
          value={last_name}
        />
        <MDBBtn onClick={updateUser}>Save Changes</MDBBtn>
      </div>
    </>
  )
}

AccountUserForm.propTypes = {
  auth: PropTypes.object.isRequired,
  updateAuthUser: PropTypes.func.isRequired
}

export default AccountUserForm
