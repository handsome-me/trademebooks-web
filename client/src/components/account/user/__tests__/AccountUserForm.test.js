import React from 'react'
import renderer from 'react-test-renderer'

import AccountUserForm from '../AccountUserForm'

describe('Component: AccountUserForm', () => {
  const props = {
    auth: {
      user: {
        first_name: 'john',
        last_name: 'doe'
      }
    },
    updateAuthUser: jest.fn()
  }

  it('should match the snapshot', () => {
    const accountUserFormSnapshot = renderer.create(<AccountUserForm {...props} />).toJSON()
    expect(accountUserFormSnapshot).toMatchSnapshot()
  })
})
