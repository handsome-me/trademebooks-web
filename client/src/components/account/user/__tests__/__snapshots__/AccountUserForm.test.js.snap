// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AccountUserForm should match the snapshot 1`] = `
Array [
  <h3
    className="mb-4 font-weight-bold"
  >
    User Settings
  </h3>,
  <div
    className="mt-5"
  >
    <h4>
      Change Name
    </h4>
    <div
      className="md-form form-group"
    >
      <i
        className="fa fa-user prefix"
        data-test="fa"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <input
        aria-disabled={false}
        className="form-control"
        data-test="input"
        disabled={false}
        name="first_name"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        required={true}
        type="text"
        value="john"
      />
      <label
        aria-labelledby=""
        className="active"
        data-error=""
        data-success=""
        id=""
        onClick={[Function]}
      >
        First Name
      </label>
    </div>
    <div
      className="md-form form-group"
    >
      <i
        className="fa fa-user prefix"
        data-test="fa"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <input
        aria-disabled={false}
        className="form-control"
        data-test="input"
        disabled={false}
        name="last_name"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        required={true}
        type="text"
        value="doe"
      />
      <label
        aria-labelledby=""
        className="active"
        data-error=""
        data-success=""
        id=""
        onClick={[Function]}
      >
        Last Name
      </label>
    </div>
    <button
      className="btn-default btn Ripple-parent"
      data-test="button"
      onClick={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      Save Changes
      <div
        className="Ripple "
        data-test="waves"
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        }
      />
    </button>
  </div>,
]
`;
